plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2DependencyHandler.kt

```diff
@@ -83,7 +83,13 @@ internal class Yarn2DependencyHandler(
 
 internal val PackageInfo.isProject: Boolean get() = value.substringAfterLast("@").startsWith("workspace:")
 
-internal val PackageInfo.moduleName: String get() = value.substringBeforeLast("@")
+internal val PackageInfo.moduleName: String get() {
+    val components = value.split(":")
+    // TODO: Handle patched packages different than non-patched ones.
+    // Patch packages have locators as e.g. the following, where the first component ends with "@patch".
+    // resolve@patch:resolve@npm%3A1.22.8#optional!builtin<compat/resolve>::version=1.22.8&hash=c3c19d
+    return components.first().substringBeforeLast("@")
```

Comment:
If only the first component is ever used, couldn't we just use `substringBefore(":")` here?

Reply:
Hmm. How about then at least specifying `limit = 2` in line 87, and probably also use the char-variant of `split`? I.e. `val components = value.split(':', limit = 2)`?

Reply:
hm, getting too complicated for my taste, so I went for your first proposal.